libvcx/src/connection.rs

```diff
@@ -908,14 +905,14 @@ impl Connection {
     }
 }
 
-pub fn update_state(handle: u32, message: Option<String>) -> VcxResult<u32> {
+pub fn update_state(handle: u32, message: Option<A2AMessage>) -> VcxResult<u32> {
+    trace!("update_state >>> handle: {}", handle);
+
     CONNECTION_MAP.get_mut(handle, |connection| {
         match connection {
-            Connections::V1(ref mut connection) => {
-                connection.update_state(message.clone())
-            }
+            Connections::V1(_) => Err(VcxError::from(VcxErrorKind::ActionNotSupported)),
             Connections::V3(ref mut connection) => {
-                connection.update_state(message.as_ref().map(String::as_str))?;
+                connection.update_state(message.clone())?;
```

Comment:
It seems that the `message` could be just moved into `update_state`?

Reply:
Used a reference instead since references should be more lightweight.